src/main/java/de/zalando/aruha/nakadi/controller/EventStreamController.java

```diff
@@ -88,6 +95,11 @@ public StreamingResponseBody streamEvents(
 
                 // validate parameters
                 if (!topicRepository.topicExists(topic)) {
+                    Optional<EventType> eventTypeO = eventTypeRepository.findByNameO(eventTypeName);
+                    if (eventTypeO.isPresent()) {
+                        writeProblemResponse(response, outputStream,INTERNAL_SERVER_ERROR, "topic is absent in kafka");
+                        return;
+                    }
```

Comment:
I agree with you, but I think we will be able to merge it when @rcillo will come back.